From 352e140832b21fee22d6e0bab929c39b744dd340 Mon Sep 17 00:00:00 2001
From: Lu Jialin <lujialin4@huawei.com>
Date: Wed, 12 Apr 2023 10:06:50 +0800
Subject: [PATCH 0896/1646] [Huawei] psi: support irq.pressure under cgroup v1

Offering: HULK
HULK inclusion
category: feature
bugzilla: 188663

-------------------------------

Export "irq.pressure" to cgroup v1 "cpuacct" subsystem.

Signed-off-by: Lu Jialin <lujialin4@huawei.com>
---
 kernel/cgroup/cgroup.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/kernel/cgroup/cgroup.c b/kernel/cgroup/cgroup.c
index bf8aaae20e82..806741a68aee 100644
--- a/kernel/cgroup/cgroup.c
+++ b/kernel/cgroup/cgroup.c
@@ -3795,6 +3795,16 @@ struct cftype cgroup_v1_psi_files[] = {
 		.poll = cgroup_pressure_poll,
 		.release = cgroup_pressure_release,
 	},
+#ifdef CONFIG_IRQ_TIME_ACCOUNTING
+	{
+		.name = "irq.pressure",
+		.flags = CFTYPE_NO_PREFIX,
+		.seq_show = cgroup_irq_pressure_show,
+		.write = cgroup_irq_pressure_write,
+		.poll = cgroup_pressure_poll,
+		.release = cgroup_pressure_release,
+	},
+#endif
 	{ } /* terminate */
 };
 #endif
-- 
2.17.1

